x-pack/plugins/transform/public/app/sections/create_transform/components/step_define/step_define_form.tsx

```diff
@@ -216,15 +270,48 @@ export const StepDefineForm: FC<StepDefineFormProps> = React.memo((props) => {
           </EuiFormRow>
         )}
 
+        {hasValidTimeField && (
+          <EuiFormRow
+            fullWidth
+            label={i18n.translate('xpack.transform.stepDefineForm.datePickerLabel', {
+              defaultMessage: 'Time range',
+            })}
+          >
+            <EuiFlexGroup alignItems="flexStart" justifyContent="spaceBetween">
+              {/* Flex Column #1: Date Picker */}
+              <EuiFlexItem>
+                <DatePickerWrapper
+                  isAutoRefreshOnly={!hasValidTimeField}
+                  showRefresh={!hasValidTimeField}
+                  width="full"
+                />
+              </EuiFlexItem>
+              {/* Flex Column #2: Apply-To-Config option */}
+              <EuiFlexItem grow={false} style={{ width: advancedEditorsSidebarWidth }}>
+                {ALLOW_TIME_RANGE_ON_TRANSFORM_CONFIG && (
```

Comment:
When would `ALLOW_TIME_RANGE_ON_TRANSFORM_CONFIG` ever be true?

Reply:
This is a hard coded feature flag at the moment so the functionality is never exposed. We'll revisit this in a follow up.